core/templates/domain/state/StateObjectFactory.ts

```diff
@@ -181,15 +181,17 @@ export class StateObjectFactory {
       written_translations: newStateTemplate.written_translations,
       next_content_id_index: newStateTemplate.next_content_id_index
     });
-    if (newState.interaction.defaultOutcome !== null) {
+    if (newState.interaction.defaultOutcome !== null &&
+        newStateName !== null) {
```

Comment:
```suggestion
    if (
      newState.interaction.defaultOutcome !== null &&
      newStateName !== null
    ) {
```

Reply:
The state name being null denotes the state name in the temporary state object created. @vojtechjelinek